### PR TITLE
Feature/dcc 5576 server error handling

### PR DESCRIPTION
**Pull request by rtisma**
- added colourized output via JANSI (added dependency)
- updated singTest.sh script to use SING_EXE variable, and so its run directory independant
- updated RestClient put() method to output a status (still needs fixing)
- added ErrorHandler to RestClient, and added try-catches where neccessary
- updated publish controller and service to return ResponseEntity<String> instead of string (so we have control over the HttpStatus)
- Added exceptions package with various modules for error handling via exception interception
- added appropriate error responses (not exceptions) where neccessary, and for uncaught exceptions, the ErrorHandler will automatically intercept and send an error response with exception message, stacktrace, requestUrl and httpStatus code
- updated profiles
- added SongError conversion from/to JSON

### CODE REVIEW

File: singTest.sh [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
-sing status -u $u | jq -C .state 
-a=`sing save -u $u`
+${SING_EXE} status -u $u | jq -C .state 
+a=`$SING_EXE save -u $u`
 echo "Got analysis id '$a'"
-sing status -u $u | jq -C .state 
-sing manifest -a $a -f manifest.txt
+${SING_EXE} status -u $u | jq -C .state 
+${SING_EXE} manifest -a $a -f manifest.txt
 cat manifest.txt
```
Comment (KevinFHartmann) [inline singTest.sh]: 👍

File: song-client/src/main/java/org/icgc/dcc/song/client/errors/ServerErrors.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+@Getter
+@RequiredArgsConstructor
+public enum ServerErrors implements ServerError {
+  TOKEN_UNAUTHORIZED("token.unauthorized"),
+  SAVE_CONFLICT("save.conflict"),
+  UNKOWN("unknown");
+
+  @NonNull private final String id;
+
+}
```
Comment (KevinFHartmann) [inline song-client/src/main/java/org/icgc/dcc/song/client/errors/ServerErrors.java]: Change "UNKOWN" to "UNKNOWN".

**Comment by rtisma** [comment]
@KevinFHartmann added one more feature, which was SongError conversion from/to Json. There is a test for this (ErrorTest.java)

**Comment by rtisma** [comment]
ok to test

**Pull request merged**